compiler/rustc_trait_selection/src/traits/coherence.rs

```diff
@@ -952,6 +952,41 @@ where
                 }
             }
 
+            ty::Alias(ty::Opaque, _) => {
```

Comment:
shouldn't we now be able to merge the `ty::Alias` branches for all variants?

I cannot imagine that we'll ever want to treat some alias differently.

I think we may just want a general comment for all aliases:
- an alias we were unable to normalize, it may normalize to anything
- if it references an inference variable/placeholder, it may normalize to that, so we have to treat it as an uncovered parameter
- if not, it may normalize to any non-generic type, this means it may be either a non-local or a local type.